### ISSUE TITLE
Benefits of CMF vs JSON-LD Vocabulary

### ISSUE THREAD

**Issue by jgmikael** [niemopen/common-model-format#2]
Hello, as a long time follower and fan of NIEM (since 2005) I'm curious about the choice of a new semantic modeling paradigm - or this is how I understand CMF (it supposedly loosens the tight coupling of semantic definitions from the technical implementation of the syntax (XML Schema). This can be seen as a improvement since it enables the use of new technical formats for the actual information exchange messages, previously being only XML documents. 

In other information exchange standardization initiatives like the EDI3.org dealing with UN/CEFACT reference data models like "Buy Ship Pay" the choice of implementation methodology has been JSON-LD. The EU has also published its Core Vocabularies as RDF and JSON-LD context (in addition to the old UML approach). Upcoming new information exchange paradigms like Trust over IP are also based on defining the exchangeable verifiable credentials according to W3C specifications, meaning in this case JSON-LD. 

Could you explain the choice of CMF in this context?

**Comment by iamdrscott** [comment]
Thanks for your question.  It's encouraging to see people paying attention!

NIEM has always used XML Schema as its modeling formalism.  The NIEM Naming and Design Rules (NDR) provide the needed layer of meaning on top of XSD, originally as text references to the RDF conceptual model, and lately in the form of specific RDF triples entailed by various XSD constructs in a NIEM schema.  All of our users are accustomed to working with models in XSD.

CMF is a NIEM message specification (or IEPD) for the information content of NIEM models expressed as XSD.  We are eating our own dog food, by first asking "what is the information content of a NIEM model?", then constructing a NIEM-conforming schema document pile for representing that information as NIEM XML.  A CMF message (or iEP) is equivalent to a NIEM schema in XSD.  There will be open source tooling to translate CMF to NIEM XSD, and vice versa.  This supports either a leisurely transition away from XSD, or an indefinite period of coexistence – whichever the future turns out to be.

NIEM already uses JSON-LD as its JSON serialization.  Any NIEM-conforming XML message can be converted to JSON-LD (and back again).  That includes NIEM models in CMF, of course.

The former NTAC (pre-OASIS) did think about dropping support for NIEM XSD and jumping straight to RDF.  We decided that change would be too large for our current users and for our release schedule.

TL;DR:  CMF is an intermediate step between the XSD status quo and a possible RDF future.
